source/MaterialXRuntime/Private/Commands/PvtRemoveMetadataCmd.cpp

```diff
@@ -0,0 +1,111 @@
+//
+// TM & (c) 2020 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved.  See LICENSE.txt for license.
+//
+
+#include <MaterialXRuntime/Private/Commands/PvtRemoveMetadataCmd.h>
+#include <MaterialXRuntime/Private/PvtCommand.h>
+
+#include <MaterialXRuntime/RtPrim.h>
+#include <MaterialXRuntime/RtTypeDef.h>
+#include <MaterialXRuntime/RtValue.h>
+
+namespace MaterialX
+{
+
+PvtRemoveMetadataCmd::PvtRemoveMetadataCmd(RtObject& obj, const RtToken& name)
+    : PvtCommand()
+    , _obj(obj)
+    , _name(name)
+    , _oldValue()
+{
+}
+
+PvtCommandPtr PvtRemoveMetadataCmd::create(RtObject& obj, const RtToken& name)
+{
+    return std::make_shared<PvtRemoveMetadataCmd>(obj, name);
+}
+
+void PvtRemoveMetadataCmd::execute(RtCommandResult& result)
+{
+    if (_obj.isValid())
+    {
+        try
+        {
+            // Send message that the metadata is being removed
+            msg().sendRemoveMetadataMessage(_obj, _name);
+
+            // Save old value for undo/redo
+            _oldValue = RtValue::clone(RtType::STRING, _obj.getMetadata(_name)->getValue(), _obj.getParent());
```

Comment:
I think you need to check that the metadata exists before accessing the value here, `_obj.getMetadata(_name)` might return nullptr.

Reply:
You should check this before sending the message. If metadata with this name doesn't exists you should probably throw an error. We do the same for removing a prim that doesn't exist.

Reply:
Do I throw an error or just return false with a message?